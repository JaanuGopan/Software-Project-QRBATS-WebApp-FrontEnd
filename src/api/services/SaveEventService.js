import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class SaveEventService {
  static async saveEvent(
    eventName,
    eventDate,
    eventValidDate,
    eventTime,
    eventEndTime,
    eventVenue,
    eventRole,
    eventModuleName,
    eventAssignedUserId
  ) {
    try {
      const response = await axios.post(ApiConstants.createEventUrl, {
        eventName: eventName,
        eventDate: eventDate,
        eventValidDate: eventValidDate,
        eventTime: eventTime,
        eventEndTime: eventEndTime,
        eventVenue: eventVenue,
        eventRole: eventRole,
        eventModuleName: eventModuleName,
        eventAssignedUserId: eventAssignedUserId,
      });
      return response.data;
    } catch (error) {
      throw new Error("Event is not created.", error);
    }
  }
}

export default SaveEventService;
