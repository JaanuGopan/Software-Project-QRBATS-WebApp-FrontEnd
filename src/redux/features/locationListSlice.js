import { createSlice } from "@reduxjs/toolkit";

export const locationListSlice = createSlice({
  name: "locationList",
  initialState: ["NCC", "LT1", "LT2", "Auditorium", "DEIE", "DMME", "DCEE"],
  reducers: {
    setLocationList: (state, action) => {
      state = [];
      state.push(...state, action.payload);
    },
    resetLocationList: (state) => {
      state = [];
    },
  },
});

export const { setLocationList, resetLocationList } = locationListSlice.actions;
export default locationListSlice.reducer;
