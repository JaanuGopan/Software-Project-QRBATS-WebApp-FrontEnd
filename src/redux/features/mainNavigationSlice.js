import { createSlice } from "@reduxjs/toolkit";

export const mainNavigationSlice = createSlice({
  name: "mainNavigation",
  initialState: {
    sideBarIndex:
      localStorage.getItem("sideBarIndex") != null
        ? parseInt(localStorage.getItem("sideBarIndex"))
        : 0,
  },
  reducers: {
    setSideBarIndex: (state, action) => {
      state.sideBarIndex = action.payload;
      localStorage.setItem("sideBarIndex", action.payload);
    },
    resetSideBarIndex: (state) => {
      state.mainNavigation = 0;
      localStorage.removeItem("sideBarIndex");
    },
  },
});

export const { setSideBarIndex, resetSideBarIndex } =
  mainNavigationSlice.actions;
export default mainNavigationSlice.reducer;
