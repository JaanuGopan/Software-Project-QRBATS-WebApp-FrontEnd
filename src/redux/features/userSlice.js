import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
  name: 'user',
  initialState: {
    user: JSON.parse(localStorage.getItem('user')) || null,
    isLoggedIn: !!localStorage.getItem('authToken'),
    authToken: localStorage.getItem('authToken') || null,
  },
  reducers: {
    login: (state, action) => {
      state.user = action.payload;
    },
    logout: (state) => {
      state.user = null;
      localStorage.removeItem('user');
      localStorage.removeItem('authToken');
    },
  },
});

export const { login, logout } = userSlice.actions;
export const selectUser = (state) => state.user.user;
export default userSlice.reducer;
export const selectIsLoggedIn = (state) => state.user.isLoggedIn;
