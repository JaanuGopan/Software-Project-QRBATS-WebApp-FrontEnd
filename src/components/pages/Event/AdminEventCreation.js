import "../Event/EventCreation/EventCreation.css";
import eventCreationImage from "../../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import QRCode from "react-qr-code";
import axios from "axios";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast from "react-hot-toast";
import { useSelector } from "react-redux";
import { selectUser } from "../../../redux/features/userSlice";

const AdminEventCreation = ({ handlecloseCreateEventWindow }) => {
  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState(null);
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventEndTime, setEventEndTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState("EVENT");
  const [eventAssignedUserId, setEventAssignedUserId] = useState(null);

  const [showModuleNameInput, setShowModuleNameInput] = useState(true);
  const [title, setTitle] = useState("Event");

  const qrCodeRef = useRef(null);

  const notifySuccess = () => toast.success("Successfully Event Created!");

  const handleEreseValu = () => {
    setEventName("");
    setModuleName(null);
    setEventDate("");
    setEventValidDate("");
    setEventTime("");
    setEventVenue("");
    setEventEndTime("");
    setEventAssignedUserId(null);
    setEventRole("EVENT");
  };

  const user = useSelector(selectUser);
  // Destructure user object for cleaner code
  const { userId } = user || {};

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await axios.post(
        "http://localhost:8080/api/v1/event/create",
        {
          eventName: eventName,
          eventDate: eventDate,
          eventValidDate: eventValidDate,
          eventTime: eventTime,
          eventEndTime: eventEndTime,
          eventVenue: eventVenue,
          eventRole: eventRole,
          eventModuleName: moduleName,
          eventAssignedUserId: userId,
        }
      );
      const responseEventName = response.data.eventName;
      console.log("Event is : " + responseEventName);
      notifySuccess();
    } catch (error) {
      console.error("Event failed", error);
    } finally {
      setShowQRCode(true);
    }
  };

  const handleDownloadQRCode = () => {
    const canvas = qrCodeRef.current.querySelector("canvas");
    const url = canvas
      .toDataURL("image/png")
      .replace("image/png", "image/octet-stream");
    const link = document.createElement("a");
    link.download = "qr_code.png";
    link.href = url;
    link.click();
  };

  const handleShareQRCode = () => {
    html2canvas(qrCodeRef.current).then((canvas) => {
      const imgData = canvas.toDataURL("image/png");
      const pdf = new jsPDF();
      pdf.addImage(imgData, "PNG", 0, 0);
      pdf.save("qr_code.pdf");
    });
  };

  // Concatenate all event details into a single string
  const eventDetails = `${eventName}, ${moduleName}, ${eventDate}, ${eventValidDate}, ${eventTime}, ${eventVenue}, ${eventAssignedUserId}`;
  const [qrCodeWindow, setQrCodeWindow] = useState(false);

  return (
    <div className="event-main-container1">
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateEventWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <h2>Create Event</h2>
      <div className="eventCreation-field">
        <img src={eventCreationImage} className="Create-logo" alt="Logo" />
        <div className="eventCreation-input-field">
          <form onSubmit={handleSubmit}>
            <div className="input-with-icon">
              <input
                type="text"
                id="eventName"
                name="eventName"
                placeholder={"Event Name"}
                className="form-control mb-2"
                value={eventName}
                onChange={(e) => setEventName(e.target.value)}
              />
            </div>
            <div className="date-div">
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    Event Starting Date
                  </label>
                  <input
                    type="date"
                    value={eventDate}
                    onChange={(e) => setEventDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    Event Ending Date
                  </label>
                  <input
                    type="date"
                    value={eventValidDate}
                    onChange={(e) => setEventValidDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="date-label" htmlFor="eventDate">
                  Venue
                </label>
                <select
                  value={eventVenue}
                  onChange={(e) => setEventVenue(e.target.value)}
                  className="form-control mb-2"
                >
                  <option value="">Select Venue</option>
                  <option value="LT1">LT1</option>
                  <option value="LT2">LT2</option>
                  <option value="Auditorium">Auditorium</option>
                  <option value="NCC">NCC</option>
                  <option value="Other">Other</option>
                </select>
              </div>
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="form-label" htmlFor="eventDate">
                  Event Starting Time
                </label>
                <input
                  type="time"
                  value={eventTime}
                  onChange={(e) => setEventTime(e.target.value)}
                  placeholder="Event Time"
                  className="form-control mb-2"
                />
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="form-label" htmlFor="enevtEndTime">
                    Event Ending Time
                  </label>
                  <input
                    type="time"
                    value={eventEndTime}
                    onChange={(e) => setEventEndTime(e.target.value)}
                    placeholder="Event End Time"
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>

            <div className="eventCreation-form"></div>
            <button
              onClick={() => setQrCodeWindow(true)}
              type="submit"
              className="btn btn-primary w-100"
            >
              Create Event
            </button>
          </form>
        </div>
        {qrCodeWindow && (
          <div className="Admin-Create-Event-Dashboard">
            {showQRCode && (
              <div ref={qrCodeRef} className="event-main-container1">
                <div
                  className="closeCreateEventWindow"
                  onClick={handlecloseCreateEventWindow}
                >
                  <IoMdCloseCircleOutline />
                </div>
                <h2>Successfully Event Created</h2>
                <div className="row-center">
                  <QRCode name="QRCode" value={eventDetails} className="mb-2" />
                </div>
                <div>
                  <p className="text-center">
                    Scan this QR code to join {title}
                  </p>
                </div>
                <div className="row-center">
                  <div className="QRbutton">
                    <button
                      onClick={handleShareQRCode}
                      className="btn btn-success mr-3"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default AdminEventCreation;
