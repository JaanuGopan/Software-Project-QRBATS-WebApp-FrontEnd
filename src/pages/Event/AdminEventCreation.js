import "./EventCreation/EventCreation.css";
import eventCreationImage from "../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import axios from "axios";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import QRCode from "qrcode.react";
import SaveEventService from "../../api/services/SaveEventService";

const AdminEventCreation = ({
  handlecloseCreateEventWindow,
  reloadEventList,
}) => {
  const [eventId, setEventId] = useState("");
  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState(null);
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventEndTime, setEventEndTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState("EVENT");
  const [eventAssignedUserId, setEventAssignedUserId] = useState(null);

  const [showModuleNameInput, setShowModuleNameInput] = useState(true);
  const [title, setTitle] = useState("Event");

  const qrCodeRef = useRef(null);

  const venueList = [
    "NCC",
    "LT1",
    "LT2",
    "Auditorium",
    "DEIE",
    "DMME",
    "DCEE",
    "Other",
  ];

  const notifySuccess = () => toast.success("Successfully Event Created!");

  const clearEventDetails = () => {
    setEventName("");
    setEventDate("");
    setEventValidDate("");
    setEventTime("");
    setEventEndTime("");
    setEventVenue("");
    setEventAssignedUserId(null);
    setEventRole("EVENT");
  };

  const user = useSelector(selectUser);
  // Destructure user object for cleaner code
  const { userId } = user || {};

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = SaveEventService.saveEvent(
        eventName,
        eventDate,
        eventValidDate,
        eventTime,
        eventEndTime,
        eventVenue,
        eventRole,
        moduleName,
        userId
      );
      setEventId(response.data.eventId);
      const responseEventName = response.data.eventName;
      notifySuccess();
      reloadEventList();
    } catch (error) {
      console.error("Event failed", error);
    } finally {
      setShowQRCode(true);
    }
  };

  const downloadQRCode = () => {
    const qrCodeURL = document
      .getElementById("qrCodeEl")
      .toDataURL("image/png")
      .replace("image/png", "image/octet-stream");
    let aEl = document.createElement("a");
    aEl.href = qrCodeURL;
    aEl.download = "QR_Code.png";
    document.body.appendChild(aEl);
    aEl.click();
    document.body.removeChild(aEl);
  };
  const eventDetails = {
    eventId: eventId,
    eventName: eventName,
    moduleName: moduleName,
    eventDate: eventDate,
    eventValidDate: eventValidDate,
    eventTime: eventTime,
    eventEndTime: eventEndTime,
    eventVenue: eventVenue,
    eventAssignedUserId: userId,
  };
  const qrCodeDetails = JSON.stringify(eventDetails);

  const handleShareQRCode = () => {
    html2canvas(qrCodeRef.current).then((canvas) => {
      const imgData = canvas.toDataURL("image/png");
      const pdf = new jsPDF();
      pdf.addImage(imgData, "PNG", 0, 0);
      pdf.save("qr_code.pdf");
    });
  };

  const [qrCodeWindow, setQrCodeWindow] = useState(false);

  return (
    <div className="event-main-container1">
      <Toaster />
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateEventWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <h2>Create Event</h2>
      <div className="eventCreation-field">
        <img src={eventCreationImage} className="Create-logo" alt="Logo" />
        <div className="eventCreation-input-field">
          <form onSubmit={handleSubmit}>
            <div className="input-with-icon">
              <input
                required
                type="text"
                id="eventName"
                name="eventName"
                placeholder={"Event Name"}
                className="form-control mb-2"
                value={eventName}
                onChange={(e) => setEventName(e.target.value)}
              />
            </div>
            <div className="date-div">
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    Event Starting Date
                  </label>
                  <input
                    required
                    type="date"
                    value={eventDate}
                    onChange={(e) => setEventDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    Event Ending Date
                  </label>
                  <input
                    required
                    type="date"
                    value={eventValidDate}
                    onChange={(e) => setEventValidDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="date-label" htmlFor="eventDate">
                  Venue
                </label>
                <select
                  required
                  value={eventVenue}
                  onChange={(e) => setEventVenue(e.target.value)}
                  className="form-control mb-2"
                >
                  <option value="">Select Venue</option>
                  {venueList.map((option, index) => (
                    <option key={index} value={option}>
                      {option}
                    </option>
                  ))}
                </select>
              </div>
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="form-label" htmlFor="eventDate">
                  Event Starting Time
                </label>
                <input
                  required
                  type="time"
                  value={eventTime}
                  onChange={(e) => setEventTime(e.target.value)}
                  placeholder="Event Time"
                  className="form-control mb-2"
                />
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="form-label" htmlFor="enevtEndTime">
                    Event Ending Time
                  </label>
                  <input
                    required
                    type="time"
                    value={eventEndTime}
                    onChange={(e) => setEventEndTime(e.target.value)}
                    placeholder="Event End Time"
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>

            <div className="eventCreation-form"></div>
            <button
              onClick={() => setQrCodeWindow(true)}
              type="submit"
              className="btn btn-primary w-100"
            >
              Create Event
            </button>
          </form>
        </div>
        {qrCodeWindow && showQRCode && (
          <div className="Admin-Create-Event-Dashboard">
            <div ref={qrCodeRef} className="event-main-container1">
              <div
                className="closeCreateEventWindow"
                onClick={handlecloseCreateEventWindow}
              >
                <IoMdCloseCircleOutline />
              </div>
              <h2>Successfully Event Created</h2>
              <div className="row-center">
                <QRCode
                  name="QRCode"
                  size={200}
                  id="qrCodeEl"
                  value={qrCodeDetails}
                  className="mb-2"
                />
              </div>
              <div>
                <p className="text-center">Scan this QR code to join {title}</p>
              </div>
              <div className="row-center">
                <div className="QRbutton">
                  <button
                    onClick={downloadQRCode}
                    className="btn btn-success mr-3"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default AdminEventCreation;
