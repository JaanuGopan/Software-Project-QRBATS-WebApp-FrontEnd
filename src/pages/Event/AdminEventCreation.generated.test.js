import renderer from 'react-test-renderer';

import eventCreationImage from "../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import QRCode from "qrcode.react";
import SaveEventService from "../../api/services/SaveEventService";
import AdminEventCreation from "./AdminEventCreation";

jest.mock("./EventCreation/EventCreation.css");
jest.mock("../../assets/Images/signin/Signin.jpeg");
jest.mock("html2canvas");
jest.mock("jspdf");
jest.mock("react-icons/io");
jest.mock("react-hot-toast");
jest.mock("react-redux");
jest.mock("../../redux/features/userSlice");
jest.mock("qrcode.react");
jest.mock("../../api/services/SaveEventService");

const renderTree = tree => renderer.create(tree);
describe('<AdminEventCreation>', () => {
  it('should render component', () => {
    expect(renderTree(<AdminEventCreation 
    />).toJSON()).toMatchSnapshot();
  });
  
});