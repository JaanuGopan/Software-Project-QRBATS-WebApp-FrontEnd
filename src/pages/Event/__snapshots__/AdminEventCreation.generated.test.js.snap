// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminEventCreation> should render component 1`] = `
<div
  className="event-main-container1"
>
  <div
    className="closeCreateEventWindow"
  />
  <h2>
    Create Event
  </h2>
  <div
    className="eventCreation-field"
  >
    <img
      alt="Logo"
      className="Create-logo"
      src="Signin.jpeg"
    />
    <div
      className="eventCreation-input-field"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-with-icon"
        >
          <input
            className="form-control mb-2"
            id="eventName"
            name="eventName"
            onChange={[Function]}
            placeholder="Event Name"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="date-div"
        >
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="date-label"
                htmlFor="eventDate"
              >
                Event Starting Date
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                required={true}
                type="date"
                value=""
              />
            </div>
          </div>
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="date-label"
                htmlFor="eventDate"
              >
                Event Ending Date
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                required={true}
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="eventCreation-form"
        >
          <div>
            <label
              className="date-label"
              htmlFor="eventDate"
            >
              Venue
            </label>
            <select
              className="form-control mb-2"
              onChange={[Function]}
              required={true}
              value=""
            >
              <option
                value=""
              >
                Select Venue
              </option>
              <option
                value="NCC"
              >
                NCC
              </option>
              <option
                value="LT1"
              >
                LT1
              </option>
              <option
                value="LT2"
              >
                LT2
              </option>
              <option
                value="Auditorium"
              >
                Auditorium
              </option>
              <option
                value="DEIE"
              >
                DEIE
              </option>
              <option
                value="DMME"
              >
                DMME
              </option>
              <option
                value="DCEE"
              >
                DCEE
              </option>
              <option
                value="Other"
              >
                Other
              </option>
            </select>
          </div>
        </div>
        <div
          className="eventCreation-form"
        >
          <div>
            <label
              className="form-label"
              htmlFor="eventDate"
            >
              Event Starting Time
            </label>
            <input
              className="form-control mb-2"
              onChange={[Function]}
              placeholder="Event Time"
              required={true}
              type="time"
              value=""
            />
          </div>
          <div
            className="eventCreation-form"
          >
            <div>
              <label
                className="form-label"
                htmlFor="enevtEndTime"
              >
                Event Ending Time
              </label>
              <input
                className="form-control mb-2"
                onChange={[Function]}
                placeholder="Event End Time"
                required={true}
                type="time"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="eventCreation-form"
        />
        <button
          className="btn btn-primary w-100"
          onClick={[Function]}
          type="submit"
        >
          Create Event
        </button>
      </form>
    </div>
  </div>
</div>
`;
